portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,21 +52,45 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String userComment = (String) entity.getProperty("userComment");
```

Comment:
In Java (and only in Java), every class inherits from the Object class, which itself has a member function toString(), which returns a string representation of the object. It's best to use this whenever you need a String value. It doesn't especially matter right now, since we know this is a String object and for objects of the String class, toString() will just return the object and cast it to String for you. If you knew for sure it's always a String being returned, casting would be faster but it is less resilient to future code changes, and in user facing code like a webserver it's best to prefer reliability over speed (within reason).